lib/boat.js

```diff
@@ -13,13 +13,17 @@ Boat.prototype.draw = function(context) {
 };
 
 Boat.prototype.moveBoatLeft = function(boat) {
-  boat.x -= boat.velocity;
-  return boat;
+  if (boat.x > 0) {
+    boat.x -= boat.velocity;
+    return boat;
+  }
 };
 
-Boat.prototype.moveBoatRight = function(boat) {
-  this.x += this.velocity;
-  return this;
+Boat.prototype.moveBoatRight = function(boat, canvasWidth) {
+  if (boat.x + boat.width < canvasWidth) {
```

Comment:
Not crazy about this implementation because it requires passing canvasWidth in from the game file